akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserEvents.scala

```diff
@@ -4,47 +4,43 @@ import akka.Done
 import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.{ActorRef, ActorSystem, Behavior}
 import akka.cluster.sharding.external.ExternalShardAllocationStrategy
-import akka.cluster.sharding.typed.ClusterShardingSettings
 import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity}
 import akka.kafka.cluster.sharding.KafkaClusterSharding
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
 object UserEvents {
-  def init(system: ActorSystem[_], settings: ProcessorSettings): Future[ActorRef[Message]] = {
+  def init(system: ActorSystem[_], settings: ProcessorSettings): Future[ActorRef[Command]] = {
     import system.executionContext
-    messageExtractor(settings).map(messageExtractor => {
+    KafkaClusterSharding(settings.system).messageExtractorNoEnvelope(
+      timeout = 10.seconds,
+      topic = settings.topics.head,
+      entityIdExtractor = (msg: Command) => msg.userId,
+      settings = settings.kafkaConsumerSettings()
+    ).map(messageExtractor => {
```

Comment:
if that fails it will be silently ignored?

Reply:
The future is used in the pipetoSelf and rethrown if it fails https://github.com/akka/akka-samples/blob/minor-kafka-updaes/akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/Main.scala#L48

Reply:
ok, missed that